# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1703001483 -3600
# Node ID bc10b7559d1839a6749c1b807f13aef304ec708b
# Parent  48fd6d91b458de93210dd8194cf831440114e811
Bug 1869557 - Focus (not select) row when tree view gets focus by keyboard. r=aleca

When a tree view (thread pane or address book's cards pane) without any previous selections
receives focus by pressing F6 or Tab, this sets focus to the first visible row, instead of
selecting the first row of all as before.

Differential Revision: https://phabricator.services.mozilla.com/D196630

diff --git a/mail/base/content/widgets/tree-view.mjs b/mail/base/content/widgets/tree-view.mjs
--- a/mail/base/content/widgets/tree-view.mjs
+++ b/mail/base/content/widgets/tree-view.mjs
@@ -233,27 +233,20 @@ class TreeView extends HTMLElement {
   }
 
   handleEvent(event) {
     switch (event.type) {
       case "keyup": {
         if (
           ["Tab", "F6"].includes(event.key) &&
           this.currentIndex == -1 &&
+          this.selectedIndex == -1 &&
           this._view?.rowCount
         ) {
-          let selectionChanged = false;
-          if (this.selectedIndex == -1) {
-            this._selection.select(0);
-            selectionChanged = true;
-          }
-          this.currentIndex = this.selectedIndex;
-          if (selectionChanged) {
-            this.onSelectionChanged();
-          }
+          this.currentIndex = this.#firstVisibleRowIndex;
         }
         break;
       }
       case "click": {
         if (event.button !== 0) {
           return;
         }
 
